runtime/src/iree/base/CMakeLists.txt

```diff
@@ -6,6 +6,48 @@
 
 iree_add_all_subdirs()
 
+iree_cc_library(
+  NAME
+    core_headers
+  HDRS
+    "alignment.h"
+    "attributes.h"
+    "config.h"
+    "target_platform.h"
+    "time.h"
+    "tracing.h"
+  PUBLIC
+)
+
+# Users providing their own allocators can have their sources live out-of-tree.
+# Note that with out of tree sources they must be absolute paths in order to be
+# found in this directory.
+set(IREE_ALLOCATOR_LIBC_SRCS "allocator_libc.c")
```

Comment:
If we have multiple out of tree sources, we could also add a CMake variable to define the external root and then allow paths to be relative to that root.